modules/runtime/nuxeo-runtime-jtajca/src/main/java/org/nuxeo/runtime/transaction/TransactionHelper.java

```diff
@@ -216,6 +216,36 @@ public static boolean isTransactionTimedOut() {
         }
     }
 
+
+    /**
+     * Gets the transaction TTL.
+     *
+     * @return the time to live in millis or {@code -1} when outside of a transaction.
+     * @since 11.5
+     */
+    public static long getTransactionTimeToLiveMs() {
```

Comment:
`startTransaction(int)` uses seconds as a unit (because that's what `TransactionManager.setTransactionTimeout` uses) so for consistency I would prefer if the TTL was returned in seconds, and name this method just `getTransactionTimeToLive`.

Reply:
yes but this is a different usage and ms could be important

Reply:
I'm not sure why milliseconds are important.
Note that even in Geronimo's `TransactionImpl` the `commit()` transaction timeout check is implemented with ` if (TransactionTimer.getCurrentTime() > timeout)` and the current time in `TransactionTimer` is updated only every second, there's no finer granularity.
So we have to consider all timeouts and TTLs as an approximation up to the second.

Reply:
it matters because in one second we could run thousands of queries and rounding 499ms to 0 can be a problem.
Now if geronimo has a second granularity I will see that next week.

Reply:
But all this happens only if we're at the last second of the original transaction timeout anyway, so if it fails half a second earlier or later it doesn't matter.